casper/src/main/scala/coop/rchain/casper/util/rholang/costacc/SlashDeploy.scala

```diff
@@ -1,16 +1,21 @@
 package coop.rchain.casper.util.rholang.costacc
 
-import com.google.protobuf.ByteString
+import coop.rchain.casper.protocol.{DeployData, DeployDataProto}
+import coop.rchain.casper.protocol.DeployData.{fromProto, toProto}
 import coop.rchain.casper.util.rholang.{SystemDeploy, SystemDeployFailure, SystemDeployUserError}
 import coop.rchain.crypto.PublicKey
-import coop.rchain.crypto.codec.Base16
 import coop.rchain.crypto.hash.Blake2b512Random
 import coop.rchain.models.BlockHash._
 import coop.rchain.models.NormalizerEnv.{Contains, ToEnvMap}
 import coop.rchain.rholang.interpreter.RhoType._
-
-final class SlashDeploy(invalidBlockHash: BlockHash, pk: PublicKey, rand: Blake2b512Random)
-    extends SystemDeploy(rand) {
+import coop.rchain.shared.Serialize
+import scodec.bits.ByteVector
+
+final case class SlashDeploy(
+    invalidBlockHash: BlockHash,
+    pk: PublicKey,
+    override val rand: Blake2b512Random
```

Comment:
The `override val` was necessary since `SystemDeploy` has a val called "rand".
But now that you said that, I realized that I don't need `override val` when I use an identifier different from "rand". Thanks.